root/etc/profile.d/switch_bash_history.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# This script is intended to pair with a docker-compose volume line
+# like /data/project_dir/bash:/root/bash to result in persistent
+# project restricted bash history preservation between build container
+# invocations
+
+TRIGGER_DIR=~/bash
+
+if [ -e $TRIGGER_DIR ]; then
+  if [ -e ~/.bash_history ]; then
+    rm -f ~/.bash_history
+  fi
+  ln -sf $TRIGGER_DIR/.bash_history ~/.bash_history
```

Comment:
Should we consider expanding this to all of the .bash items in the home directory? We'd likely want to copy over .bash_profile and .bashrc if they didn't exist in $TRIGGER_DIR rather than just link like we can with .bash_history.

Specifically I've found that I desire this when working with things like Acquia's BLT toolset which sets up a blt alias in .bash_profile as an alternative to PATH manipulations to get vendor/bin pulled in.

Reply:
I think this makes a lot of sense. That seems like something we should do as a follow-up.